src/rgw/driver/rados/rgw_reshard.cc

```diff
@@ -837,7 +837,12 @@ int RGWBucketReshard::do_reshard(const rgw::bucket_index_layout_generation& curr
     while (is_truncated) {
       entries.clear();
       int ret = store->getRados()->bi_list(dpp, bucket_info, i, null_object_filter, marker, max_entries, &entries, &is_truncated);
-      if (ret < 0 && ret != -ENOENT) {
+      if (ret == -ENOENT) {
```

Comment:
wow, seems like;  have we observed this inf loop?

Reply:
i was seeing `ENOENT` here during testing because it was trying to read from the wrong object name. i hadn't seen it happen before